src/presentation/services/device-event-service.ts

```diff
@@ -0,0 +1,32 @@
+import { DeviceCreatedEvent, DeviceEvent } from '../value-objects/device-event'
+import { DeviceEventEmitter } from '../../business-logic/services/interfaces/device-event-emitter'
+import { DeviceEventObserver } from '../interfaces/device-event-observer'
+import { Device } from '../../model/entities/device'
+import { DeviceEventBuilder } from '../interfaces/device-event-builder'
+
+export class DeviceEventService implements DeviceEventEmitter, DeviceEventObserver {
+
+  private static instance: DeviceEventService
+  private readonly callbacks: ((deviceEvent: DeviceEvent) => void)[] = []
+
+  constructor(private readonly deviceEventBuilder: DeviceEventBuilder) { }
+
+  public static getInstance(deviceBuilder: DeviceEventBuilder): DeviceEventService {
+    if(!this.instance) {
+      this.instance = new DeviceEventService(deviceBuilder)
+    }
+    return this.instance
```

Comment:
Trying a new way to assign instances here 😄 

```suggestion
    this.instance ??= new DeviceEventService(deviceBuilder)
    return this.instance
```